ext/dba/dba.c

```diff
@@ -717,12 +717,16 @@ static void php_dba_open(INTERNAL_FUNCTION_PARAMETERS, bool persistent)
 
 	info = pemalloc(sizeof(dba_info), persistent);
 	memset(info, 0, sizeof(dba_info));
-	info->path = pestrdup(ZSTR_VAL(path), persistent);
 	info->mode = modenr;
 	info->file_permission = permission;
 	info->map_size = map_size;
 	info->flags = (hptr->flags & ~DBA_LOCK_ALL) | (lock_flag & DBA_LOCK_ALL) | (persistent ? DBA_PERSISTENT : 0);
 	info->lock.mode = lock_mode;
+	if (persistent) {
+		info->path = zend_string_dup(path, /* persistant */ true);
```

Comment:
Use zend_string_init instead, zend_string_dup does not duplicate interned strings.

Reply:
Also persistant -> persistent